### ISSUE TITLE
Consider using the parent class name as a prefix to disambiguate names

### ISSUE THREAD

**Issue by dvdsgl** [quicktype/quicktype#266]
This was [suggested on Reddit](https://www.reddit.com/r/swift/comments/7bakaa/demo_quickly_read_json_data_as_structs_in_swift/dpjx43w/).

For a top-level sample named `Data`, `album.data` could be a type named `AlbumData` rather than `PurpleData`:

```json
{
  "album": {
    "data": { ... }
  }
}
```

**Comment by ffittschen** [comment]
For Swift, another solution to this could be nested types, i.e. take the top-level type and generate all other types within that `struct`/`class`

```swift
struct Album {
    struct Data {
        // ...
    }

    let data: Album.Data
}
```

This would also help to avoid conflicts if I have multiple JSON objects that have the same name but different top-level types.

**Comment by schani** [comment]
We're considering this for other target languages, too.  Right now we're missing the analysis to make this happen, plus it's not high on the priority list.  But we'd love to help if you'd like to contribute ;-)

**Comment by schani** [comment]
Fixed in dfa218e8dff89b44e0c026c029b822b528c9bd15
